x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/monitor_errors/errors_list.tsx

```diff
@@ -74,23 +86,21 @@ export const ErrorsList = ({
             locationId={selectedLocation?.id}
           />
         );
-        const isActive = isActiveState(item);
-        if (!isActive || lastTestRun.state.id !== item.state.id) {
-          return link;
+        if (isErrorActive(item, lastErrorTestRun?.state.id, latestPing?.monitor.status)) {
```

Comment:
I find it's a lot easier to mentally parse positive conditions, so I refactored this block to be the condition, and made the `return link` line the default. I also extracted the conditions into a function and added unit tests for it to help clarify when it should return `true` or `false`.